pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -298,26 +311,23 @@ func (c *Controller) updateOperatorStatus(workload *appsv1.Deployment, operatorC
 		c.versionRecorder.SetVersion(operandName, c.targetOperandVersion)
 	}
 
-	updateGenerationFn := func(newStatus *operatorv1.OperatorStatus) error {
+	// set updateGenerationFn so that it is invoked in defer
+	updateGenerationFn = func(newStatus *operatorv1.OperatorStatus) error {
 		resourcemerge.SetDeploymentGeneration(&newStatus.Generations, workload)
 		return nil
 	}
 
-	if _, _, updateError := v1helpers.UpdateStatus(c.operatorClient,
-		v1helpers.UpdateConditionFn(deploymentAvailableCondition),
-		v1helpers.UpdateConditionFn(deploymentDegradedCondition),
-		v1helpers.UpdateConditionFn(deploymentProgressingCondition),
-		v1helpers.UpdateConditionFn(workloadDegradedCondition),
-		updateGenerationFn); updateError != nil {
-		return updateError
-	}
-
 	if len(errs) > 0 {
 		return kerrors.NewAggregate(errs)
 	}
 	return nil
 }
 
+func isUpdatingTooLong(operatorStatus *operatorv1.OperatorStatus, progressingConditionType string) (bool, error) {
+	progressing := v1helpers.FindOperatorCondition(operatorStatus.Conditions, progressingConditionType)
+	return progressing != nil && progressing.Status == operatorv1.ConditionTrue && time.Now().After(progressing.LastTransitionTime.Add(5*time.Minute)), nil
```

Comment:
these 5 min are over all instances, aren't they? If you deploy in a (soon) 5 node cluster with graceful termination we can can reach 5 min in normal operation can't we?

Reply:
it's per instance due to:
```
deploymentProgressingCondition.Message = fmt.Sprintf("deployment/%s.%s: %d/%d pods have been updated to the latest generation", workload.Name, c.targetNamespace,
```
which will update the progressing condition every time a new pod was updated to the latest version

Reply:
add a short comment please. This is not obvious from the code.